onebusaway-android/src/main/java/org/onebusaway/android/ui/TripDetailsActivity.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2012-2015 Paul Watts (paulcwatts@gmail.com), University of South Florida
+ * and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.onebusaway.android.ui;
+
+import org.onebusaway.android.io.ObaAnalytics;
+import org.onebusaway.android.provider.ObaContract;
+import org.onebusaway.android.util.FragmentUtils;
+import org.onebusaway.android.util.UIHelp;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v4.app.FragmentManager;
+import android.support.v7.app.ActionBarActivity;
+import android.view.MenuItem;
+
+public class TripDetailsActivity extends ActionBarActivity {
+
+    public static void start(Context context, String tripId, String stopId) {
+        context.startActivity(makeIntent(context, tripId, stopId));
+    }
+
+    public static Intent makeIntent(Context context, String tripId, String stopId) {
+        Intent myIntent = new Intent(context, TripDetailsActivity.class);
+        myIntent.setData(ObaContract.Trips.buildUri(tripId, stopId));
```

Comment:
Because you're not actually using the `URI`, and you're just parsing it back to `tripId and stopId` in `TripDetailsListFragment.onActivityCreated()`, I think I'd prefer to just have you pass the separate Strings in the `Intent`.  And, this can be set up as an `IntentBuilder` in `TripDetailsListFragment` - you can basically copy/paste the code from `ArrivalListFragment` for this:
https://github.com/OneBusAway/onebusaway-android/blob/develop/onebusaway-android/src/main/java/org/onebusaway/android/ui/ArrivalsListFragment.java#L175

..and you'd put the `tripId`/`stopId` as a Strings in the bundle, like stop name and other fields are being handled in `ArrivalsListFragment`:
`mIntent.putExtra(ArrivalsListFragment.STOP_NAME, stopName);`